src/ServiceProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace JustBetter\StatamicMetadataTemplates;
+
+use Statamic\Providers\AddonServiceProvider;
+use JustBetter\StatamicMetadataTemplates\Fieldtypes\MetaTemplateCollections;
+use Statamic\Facades\Collection;
+use Statamic\Facades\Entry;
+use Statamic\Statamic;
+
+class ServiceProvider extends AddonServiceProvider
+{
+    public function register(): void
+    {
+        $this->registerConfig();
+    }
+
+    public function bootComputed(): self
+    {
+        if (!Statamic::isCpRoute()) {
+            $metadataFields = config('justbetter-meta-templates.fields') ?? [];
+            foreach (config('justbetter-meta-templates.collections') ?? [] as $collection) {
+                foreach ($metadataFields as $field => $addonField) {
+                    Collection::computed($collection, $field, function ($entry, $value) use ($metadataFields, $field, $addonField) {
+                        if ($value) {
+                            return $value;
+                        }
+
+                        $metaTemplate = Entry::query()->where('collection', 'meta_templates')->where('for_collection', $entry->collection->handle())->first();
+
+                        if (!$metaTemplate) {
+                            return $value;
+                        }
+
+                        $templateData = $metaTemplate->get($metadataFields[$field]);
+
+                        return str_replace(['{title}'], [$entry->title], $templateData);
```

Comment:
Maybe make this generic so all attributes can be used as variables? So if there is a field "price" you can use `{price}`. Search for those variables with regex (`/{(.*?)}/g` should do it) and replace it if it exists. So if it's an un-existing variable it just returns an empty string.